include/seqan3/search/algorithm/search.hpp

```diff
@@ -87,63 +140,47 @@ inline auto search(queries_t && queries,
                    index_t const & index,
                    configuration_t const & cfg = search_cfg::default_configuration)
 {
-    if constexpr(dimension_v<queries_t> == 1u)
+    using search_traits_t = detail::search_traits<configuration_t>;
+
+    // If no mode was set, default to search all hits.
+    if constexpr (!search_traits_t::has_mode_configuration)
     {
-        static_assert(std::ranges::random_access_range<queries_t>, "The query sequence must model random_access_range.");
-        static_assert(std::ranges::sized_range<queries_t>, "The query sequence must model sized_range.");
+        return search(std::forward<queries_t>(queries), index, cfg | search_cfg::mode{search_cfg::all});
     }
     else
-    {
-        static_assert(std::ranges::forward_range<queries_t>, "The query collection must model forward_range.");
-        static_assert(std::ranges::sized_range<queries_t>, "The query collection must model sized_range.");
-        static_assert(std::ranges::random_access_range<value_type_t<queries_t>>,
-                      "Elements of the query collection must model random_access_range.");
-        static_assert(std::ranges::sized_range<value_type_t<queries_t>>,
-                      "Elements of the query collection must model sized_range.");
-    }
-
-    static_assert(detail::is_type_specialisation_of_v<remove_cvref_t<configuration_t>, configuration>,
-                  "cfg must be a specialisation of seqan3::configuration.");
+    {   // If no output configuration was, default to returning text positions.
+        if constexpr (!search_traits_t::has_output_configuration)
+        {
+            return search(std::forward<queries_t>(queries), index, cfg | search_cfg::output{search_cfg::text_position});
+        }
+        else
+        {
+            if constexpr(dimension_v<queries_t> == 1u)
+            {
+                static_assert(std::ranges::random_access_range<queries_t>,
+                              "The query sequence must model random_access_range.");
+                static_assert(std::ranges::sized_range<queries_t>, "The query sequence must model sized_range.");
+            }
+            else
+            {
+                static_assert(std::ranges::forward_range<queries_t>, "The query collection must model forward_range.");
+                static_assert(std::ranges::sized_range<queries_t>, "The query collection must model sized_range.");
+                static_assert(std::ranges::random_access_range<value_type_t<queries_t>>,
+                            "Elements of the query collection must model random_access_range.");
+                static_assert(std::ranges::sized_range<value_type_t<queries_t>>,
+                            "Elements of the query collection must model sized_range.");
+            }
 
-    using cfg_t = remove_cvref_t<configuration_t>;
+            static_assert(detail::is_type_specialisation_of_v<remove_cvref_t<configuration_t>, configuration>,
```

Comment:
No they don't but albeit I did some little refactoring. I think the entire configuration step should be refactored. It would be great to have a separate configuration step similar to the alignment code and then invoke the configured search function. This makes it also easier to add parallel execution and possibly reusing the existing execution handler for this.